docs/Configuration_Options.md

```diff
@@ -26,8 +26,11 @@ When running `hokusai setup` the following files are created:
     - `post-build`: <string> (optional) - A post-build hook - useful for image post-processing
     - `pre-deploy`: <string> (optional) - A pre-deploy hook - useful to enforce migrations
     - `post-deploy`: <string> (optional) - A post-deploy hook - useful for deploy notifications
-    - `git-remote`: <string> (optional) - Push deployment tags to git remote when invoking the `hokusai [staging|production] deploy` or the `hokusai pipeline promote` commands.  Can either be a local alias like 'origin' or a URI like `git@github.com:artsy/hokusai.git`.  Bound to the `--git-remote` option for these commands.
-    - `run-tty`: <boolean> (optional) - Attach the terminal to your shell session when invoking `hokusai [staging|production] run`.  Bound to the `--tty` option for this command, overridden by the `HOKUSAI_RUN_TTY` env var.
+    - `git-remote`: <string> (optional) - Push deployment tags to git remote when invoking the `hokusai [staging|production|review_app] deploy` or the `hokusai pipeline promote` commands.  Can either be a local alias like 'origin' or a URI like `git@github.com:artsy/hokusai.git`.  Bound to the `--git-remote` option for these commands.
+    - `run-tty`: <boolean> (optional) - Attach the terminal to your shell session when invoking `hokusai [staging|production|review_app] run`.  Bound to the `--tty` option for this command, overridden by the `HOKUSAI_RUN_TTY` env var.
```

Comment:
@ashfurrow I think you brought this up initially, but I now have

```
export HOKUSAI_RUN_TTY=True
export HOKUSAI_TAIL_LOGS=10
export HOKUSAI_FOLLOW_LOGS=True
```

in my `~/.bashrc` so I don't have to type those `--tty` / `--follow` / `--tail` options all the time :)

Reply:
Also including

```
HOKUSAI_RUN_CONSTRAINTS=tier=background
HOKUSAI_ALWAYS_VERBOSE=True
```

Reply:
nice!

Reply:
Minor comment: The environment variables shouldn't _override_ the provided flag, but should provide an alternate default. I.e., if the default is usually `false` but the env var is set, the effective default would be `true`. However, passing a flag explicitly (`--tty` in this case) should still take precedence.

Maybe it already works that way and this is just a comment about the docs.

Reply:
I'd also expect the behavior @joeyAghion describes.

Conversely, it would be nice if you could negate a truthy environment default with `--no-tty` or `--no-tail`.

Botocore uses a chain of configuration providers, which looks pretty cool. It would be nice if that was extracted into a library of its own.

>In botocore most of our config chains follow the the precendence: 
> session_instance_variables, environment, config_file, default_value.

https://github.com/boto/botocore/blob/master/botocore/configprovider.py#L125-L126

> Each provider in the chain is called, the first one returning a non-None value is then returned.

https://github.com/boto/botocore/blob/master/botocore/configprovider.py#L295-L296